src/nvim/testdir/test_edit.vim

```diff
@@ -1534,3 +1534,28 @@ func Test_edit_noesckeys()
   bwipe!
   " set esckeys
 endfunc
+
+" Test for editing a directory
+func Test_edit_is_a_directory()
+  CheckEnglish
+  let dirname = getcwd() . "/Xdir"
+  call mkdir(dirname, 'p')
+
+  new
+  redir => msg
+  exe 'edit' dirname
+  redir END
+  call assert_match("is a directory$", split(msg, "\n")[0])
```

Comment:
Failed on Windows.

Reply:
Wow.  I don't know the right fix for Windows...

Reply:
Can you fix it?

Reply:
I think the fail is not affected with the patch.
The directory check is already broken before...

Reply:
Cherry-pick `CheckNotMSWindows` from patch 8.1.1826. Run it before or after `CheckEnglish` and leave a TODO comment.

I don't know if there is a fix. I can't fix it now.

Reply:
OK.  I have added it.